src/components/Header/tests/header.test.js

```diff
@@ -0,0 +1,85 @@
+import { combineReducers } from 'redux';
+import { Provider } from 'react-redux';
+import { BrowserRouter } from 'react-router-dom';
+import configureStore from 'redux-mock-store';
+import { render, screen } from '@testing-library/react';
+
+import coursesReducer from '../../../store/courses/reducer';
+import authorsReducer from '../../../store/authors/reducer';
+import userReducer from '../../../store/user/reducer';
+
+import Header from '../Header';
+
+const rootReducer = combineReducers({
+	courses: coursesReducer,
+	authors: authorsReducer,
+	user: userReducer,
+});
+
+const mockStore = configureStore({
+	reducer: rootReducer,
+});
+
+describe('Header Component', () => {
+	let store;
+
+	beforeEach(() => {
+		store = mockStore({
+			user: {
+				name: 'Test user',
+				email: 'admin@email.com',
+				password: 'admin123',
+				role: 'admin',
+				id: '85aa767d-73c0-4e23-9a02-27902d418b37',
+				isAuth: true,
+			},
+		});
+	});
+
+	test('Header should have logo with Company Logo alt', () => {
+		render(
+			<Provider store={store}>
```

Comment:
You can move Provider and BrowserRouter into some Wrapper component which renders children, for example
```
const Wrapper = ({ children }) => (
    <Provider store={store}>
        <BrowserRouter>
            {children}
        </BrowserRouter>
    </Provider>
);
```

and then use it like:
```
render(<Header />, { wrapper: Wrapper });
```
as you can pass wrapper component into `render` second argument